examples/Custom_Hooks/callable.py

```diff
@@ -0,0 +1,92 @@
+## E2E Test for using callables
+
+# This example shows how to use the register_hook function to request internal data from different objects such as campaign, recommenders and objective.
+
+# To request internal data from the recommendation process this function can be used by giving it two arguments:
+# - The class method that contains the required data
+# - The callable function that will process the data
+# The function wraps both functions and returns the wrapped function
+
+# This examples assumes some basic familiarity with using BayBE.
+# We refer to [`campaign`](./campaign.md) for a more general and basic example.
+
+### Necessary imports for this example
+
+
+import pandas as pd
+
+from baybe import Campaign
+from baybe.objectives import SingleTargetObjective
+from baybe.objectives.base import Objective
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.recommenders import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.targets import NumericalTarget
+from baybe.utils.basic import register_hook
+from baybe.utils.dataframe import add_fake_results
+
+### Setup
+
+# Define three test functions to test the functionality of register_hook().
+# Note that the callable function needs to have the same signature like the function from which the data are required.
+
+
+def recommend_test(
+    self,
+    batch_size: int,
+    searchspace: SearchSpace,
+    objective: Objective | None = None,
+    measurements: pd.DataFrame | None = None,
+):
+    """Print the searchspace and the batch size from the recommend call."""
+    print("start recommend_test")
```

Comment:
perhaps add some print that uses `self`, maybe not `self` itself but `self.some_property`